pennylane/compiler.py

```diff
@@ -0,0 +1,227 @@
+                theta = jnp.abs(x)
+                qml.RY(theta, wires=0)
+                qml.Rot(z[0], z[1], z[2], wires=0)
+                return qml.state()
+
+        >>> circuit(0.2j, jnp.array([0.3, 0.6, 0.9]))  # calls precompiled function
+        array([0.75634905-0.52801002j, 0. +0.j,
+            0.35962678+0.14074839j, 0. +0.j])
+
+        Catalyst also supports capturing imperative Python control flow in compiled programs. You can
+        enable this feature via the ``autograph=True`` parameter. Note that it does come with some
+        restrictions, in particular whenever global state is involved. Refer to the documentation page
+        for a complete discussion of the supported and unsupported use-cases.
+
+        .. code-block:: python
+
+            @qjit(autograph=True)
+            @qml.qnode(qml.device("lightning.qubit", wires=2))
+            def circuit(x: int):
+
+                if x < 5:
+                    qml.Hadamard(wires=0)
+                else:
+                    qml.T(wires=0)
+
+                return qml.expval(qml.PauliZ(0))
+
+        >>> circuit(3)
+        array(0.)
+
+        >>> circuit(5)
+        array(1.)
+
+        Note that imperative control flow will still work in Catalyst even when the AutoGraph feature is
+        turned off, it just won't be captured in the compiled program and cannot involve traced values.
+        The example above would then raise a tracing error, as there is no value for ``x`` yet than can
+        be compared in the if statement. A loop like ``for i in range(5)`` would be unrolled during
+        tracing, "copy-pasting" the body 5 times into the program rather than appearing as is.
+
+        .. important::
+
+            Most decomposition logic will be equivalent to PennyLane's decomposition.
+            However, decomposition logic will differ in the following cases:
+
+            1. All :class:`qml.Controlled <pennylane.ops.op_math.Controlled>` operations will decompose
+                to :class:`qml.QubitUnitary <pennylane.QubitUnitary>` operations.
+
+            2. :class:`qml.ControlledQubitUnitary <pennylane.ControlledQubitUnitary>` operations will
+                decompose to :class:`qml.QubitUnitary <pennylane.QubitUnitary>` operations.
+
+            3. The list of device-supported gates employed by Catalyst is currently different than that
+                of the ``lightning.qubit`` device, as defined by the
+                :class:`~.pennylane_extensions.QJITDevice`.
+        """
+
+        if not cls.__BACKENDS and not cls.available():
+            raise RuntimeError("There is no available compiler package.")
+
+        if "cpl_qjit" not in cls.__ENTRY_POINTS:
+            raise RuntimeError("There is no available 'cpl_qjit' entry point.")
+
+        qjit_loader = cls.__ENTRY_POINTS["cpl_qjit"].load()
+        return qjit_loader(fn=fn, *args, **kwargs)
+
+
+# Exported Methods
+
+qjit = Compiler.qjit
```

Comment:
@maliasadi yeah after reading through, I would agree with @dime10 here. I would favour module level functions and variables over a static class, alongside private functions for things you don't want public/in the docs.

This will also help with documentation etc. (I think the exporting you do here for `qjit` may not always copy the docstring as you expect)